File: 02-array-tuples/py/arrays.py
from array import array
if __name__ == "__main__":

    #typecode (1st arg) must be b, B, u, h, H, i, I, l, L, q, Q, f or d)
    arr = array('i', [1, 2, 3, 4, 5])

    # better use `numpy.zeros((500,), dtype=numpy.uint64)` 
    big_array = array('Q', [0] * 500)

    print("first element of the array:", arr[0])
    print("last element of the array:", arr[-1])

    print("small size:", len(arr))
    print("big size:", len(big_array))

    print("a section of the array as a slice", arr[1:3])

    # Going beyond the limits , i.e "out of range"
    print(arr[5])

File: 02-array-tuples/rs/src/arrays.rs
pub fn demo() {
    let mut xs = [1, 2, 3, 4, 5];

    let ys: [i32; 500] = [0; 500];
    let _ys = [0_u64; 500];

    println!("first element of the array: {}", xs[0]);
    println!("second element of the array: {}", xs[1]);

    println!("small size: {}", xs.len());
    println!("big size: {}", ys.len());

    println!("borrow a section of the array as a slice {:?}", &xs[1..3]);

    match xs.get_mut(5) {
        Some(element) => {
            *element = 0;
        }
        None => {
            println!("oops, out of bounds");
        }
    };
}
